src/server/common/controller/page-controller/page-controller.js

```diff
@@ -0,0 +1,61 @@
+import { calculateNextPage } from '../../helpers/next-page.js'
+import { ExitPage } from '../../model/page/exit-page-model.js'
+/** @import { Server, ServerRegisterPluginObject } from '@hapi/hapi' */
+/** @import { NextPage } from '../../helpers/next-page.js' */
+/** @import { RawPayload } from '../../model/answer/answer-model.js' */
+/** @import { Page } from '../../model/page/page-model.js' */
+
+export class PageController {
+  /**
+   * @param {Page} page
+   */
+  constructor(page) {
+    this.page = page
+  }
+
+  /** @returns {ServerRegisterPluginObject<void>} */
+  plugin() {
+    return {
+      plugin: {
+        name: `${this.page.sectionKey}-${this.page.key}`,
+
+        /** @param {Server} server */
+        register: (server) => {
+          server.route([
+            {
+              method: 'GET',
+              path: this.page.urlPath,
+              handler: this.getHandler.bind(this)
+            },
+            {
+              method: 'POST',
+              path: this.page.urlPath,
+              handler: this.postHandler.bind(this)
+            }
+          ])
+        }
+      }
+    }
+  }
+
+  getHandler(req, h) {
+    return h.view(this.page.view, {
+      nextPage: req.query.redirect_uri,
```

Comment:
Since we're rendering this as a url, we need to call nextPage here, rather than in the POST handler, for content pages.

Reply:
I noticed we've supplemented this with continueURL - I guess removing nextPage is the move in this case?

Reply:
more than happy to remove - but thought it might be used at some point so better to leave in over remove and put back in again - your call